maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java

```diff
@@ -523,15 +535,122 @@ private File loadProjectProperties(Builder builder, MavenProject bndProject, Mav
 		if (baseDir != null) {
 			builder.updateModified(pomFile.lastModified(), "POM: " + pomFile);
 		}
+
+		boolean bndUsed = false;
 		Xpp3Dom bndElement = configuration.getChild("bnd");
 		if (bndElement != null) {
-			logger.debug("loading bnd properties from bnd element in pom: {}", pomProject);
-			UTF8Properties properties = new UTF8Properties();
-			properties.load(bndElement.getValue(), pomFile, builder);
-			// we use setProperties to handle -include
-			builder.setProperties(baseDir, properties.replaceHere(baseDir));
+			if (projectFile == null) {
+				if (!xppDomEqualsWithSubtitution(bndElement, tracker.get("bnd"), projectFile, builder)) {
+					tracker.put("bnd", bndElement);
+					bndUsed = true;
+					logger.debug("loading bnd properties from bnd element in pom: {}", pomProject);
+					projectFile = pomFile;
+					UTF8Properties properties = new UTF8Properties();
+					properties.load(bndElement.getValue(), projectFile, builder);
+					// we use setProperties to handle -include
+					builder.setProperties(baseDir, transformUnderscoreKeys(properties.replaceHere(baseDir)));
+				}
+			} else {
+				logger.warn("Pom defines both bndfile and bnd configuration. Ignoring the bnd configuration in pom: {}",
+					pomProject);
+			}
+		}
+
+		if (instructions != null && instructions.size() > 0 && configuration.getChild("instructions") != null) {
```

Comment:
This is called once per each parent pom, but the evaluated properties field `instruction` is always the same (i.e. already considering the inheritance for this configuration parameter alone). Are you sure that the inheritance behaviour is the same for `bnd` and `instructions` (according to https://blog.sonatype.com/2011/01/maven-how-to-merging-plugin-configuration-in-complex-projects/)?

Reply:
I put tests in place that ensure the correct inheritance behaviour is happening. They are part of the pull request. If you have a more specific concert, please let me know.

Reply:
I would stop recursion in case you found instructions, cause the additional calls of `loadProjectProperties` mean additional effort but will not lead to different results IIUC.

Reply:
I think I know what you mean but I think for that loadProjectProperties would need to return some stopcondition which may make the code a little more complicated as it already returns a File. Is the slight extra time spent during the build worth the extra complexity or do you have a simple solution in mind?